Formula/file-roller.rb

```diff
@@ -22,18 +22,40 @@ class FileRoller < Formula
   depends_on "hicolor-icon-theme"
   depends_on "gnome-icon-theme"
 
+  # Add undeclared dependencies reported by `brew linkage`
+  depends_on "atk"
+  depends_on "cairo"
+  depends_on "gdk-pixbuf"
+  depends_on "gettext"
+  depends_on "glib"
+  depends_on "pango"
+
+  # These can be removed when the next version (>3.16.4) is released, since they
+  # are only required to support the patch below, which will no longer be needed
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "yelp-tools" => :build
+
+  # Remove when >3.16.4 is released
```

Comment:
Maybe something like this?

```
diff --git a/Formula/file-roller.rb b/Formula/file-roller.rb
index 9ebe38b..36c41a5 100644
--- a/Formula/file-roller.rb
+++ b/Formula/file-roller.rb
@@ -37,20 +37,19 @@ class FileRoller < Formula
   depends_on "libtool" => :build
   depends_on "yelp-tools" => :build
 
-  # Remove when >3.16.4 is released
   # Build fails during schema validation when using glib >= 1.45.1
   # Upstream patch for https://bugzilla.gnome.org/show_bug.cgi?id=756607
   patch do
     url "https://git.gnome.org/browse/file-roller/patch/?id=37df6d640817127dbad0f9ba94141cfb1e6a8360"
     sha256 "5dee926832a607fc703518cd35b8fe0846c6bf760faf19982a89fe5f5c21fe75"
+    assert_version "3.16.4"
   end
 
   def install
     ENV.append "CFLAGS", "-I#{Formula["libmagic"].opt_include}"
     ENV.append "LIBS", "-L#{Formula["libmagic"].opt_lib}"
 
-    # Remove when >3.16.4 is released
-    system "autoreconf", "-fiv"
+    assert_version("3.16.4") { system "autoreconf", "-fiv" }
 
     # forces use of gtk3-update-icon-cache instead of gtk-update-icon-cache. No bugreport should
     # be filed for this since it only occurs because Homebrew renames gtk+3's gtk-update-icon-cache

```

```
diff --git a/Library/Homebrew/formula.rb b/Library/Homebrew/formula.rb
index 3e6006b..15e1e9e 100644
--- a/Library/Homebrew/formula.rb
+++ b/Library/Homebrew/formula.rb
@@ -217,6 +217,19 @@ class Formula
     active_spec == stable
   end
 
+  def assert_version(expected, strict=true, &block)
+    matches = version == Version.new(expected)
+    unless matches
+      if strict
+        raise "verify change(s) still needed for #{version}"
+      else
+        opoo "verify change(s) still needed for #{version}"
+      end
+    end
+    yield if block_given?
+    matches
+  end
+
   # Is the currently active {SoftwareSpec} a {#devel} build?
   # @private
   def devel?
@@ -918,6 +931,11 @@ class Formula
   def patch
     unless patchlist.empty?
       ohai "Patching"
+      patchlist.each do |p|
+        if p.respond_to?(:assert_version) && !p.assert_version.nil? && Version.new(p.assert_version) != version
+          raise "verify change(s) still needed for #{version}"
+        end
+      end
       patchlist.each(&:apply)
     end
   end
diff --git a/Library/Homebrew/resource.rb b/Library/Homebrew/resource.rb
index 3fcdf46..613c952 100644
--- a/Library/Homebrew/resource.rb
+++ b/Library/Homebrew/resource.rb
@@ -185,6 +185,10 @@ class Resource
       @patch_files.concat(paths)
       @patch_files.uniq!
     end
+
+    def assert_version(v=nil)
+      v
+    end
   end
 end
 
```